File: atcoder/abc136B_uneven_numbers.nim
# unihernandez22
# https://atcoder.jp/contests/abc136/tasks/abc136_b
# implementation

from strutils import parseInt, intToStr

let n = stdin.readLine.parseInt

var ans = 0
for i in 1..n:
  if i.intToStr.len mod 2 == 1: inc ans

ans.echo

File: atcoder/abc136B_uneven_numbers.ml
(* vicfred
 * https://atcoder.jp/contests/abc136/tasks/abc136_b
 * implementation
 * *)
Scanf.scanf "%d\n" @@ fun n ->
    Printf.printf "%d\n" @@
    Array.fold_left ( + ) 0 @@
    Array.init n @@ fun i -> if (String.length @@ string_of_int @@ i + 1) mod 2 = 1 then 1 else 0

